### PR TITLE
Fix plot overplot

### LINKED ISSUE #2128: plot with overplot
This has long been an annoyance but answering

- #2119 

made me write this up.

With 4.16 we have (the two datasets are the same for this example):

```
>>> plot("data", "data", 2)
```

sherpa-1

If I try to use `overplot` then I get the following, which doesn't make much sense

```
>>> plot("model", "model", 2, overplot=True)
```

sherpa-2

So, the original data has been removed. Now, there is the confusing `clearwindow` argument, but

```
>>> plot("data", "data", 2)
>>> plot("model", "model", 2, overplot=True, clearwindow=False)
```

produces the same display.

I will note the slightly-different result from

```
>>> plot("data", "data", 2)
>>> plot("model", "model", 2, clearwindow=False)
```

sherpa-3

which does not  hide the plot axes/title.

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

Improve the behaviour of plot() and contour() when called with overplot=True or overcontour=True respectively. Fix #2128.

# Details

Fortunately with the changes made in #2126, in particular sorting out the rows/cols settings and dropping the special case of the "singleton" plot code, this turned out to be easier than  I'd thought. 

So, this PR needs #2126.

So, we need to be able to ask the SplitPlot class what

- the current number of rows and columns are
- how many plots are currently active
- not to clear the window when resetting

At present there is no actual API to do these, so we rely in access to internal fields of the class. Ideally we would add an API but

a) the UI code is basically the only real user of the SplitPlot API at the moment (the JointPlot class subplots SplitPlot but adds it's own API which means we skip this complexity for commands like plot_fit_resid)
b) it's not entirely clear what API we want, so I'd rather live with this code for now until we get more experience of what we want

I will say that the existing SplitPlot API - e.g. addplot, overplot, plot, is a bit covfusing and perhaps could be rewritten to better match the plot vs overplot needs, but I do not think it worth working on at this time.

# Examples

```
>>> plot("data", "data", 2, cols=1)
>>> plot("model", "model", 2, overplot=True)
```

sherpa-1

Now, what should happen when you supply too few plots in the overplot call - e.g.

```
>>> plot("data", "data", 2, cols=1)
>>> plot("model", overplot=True)
```
sherpa-2

For now we allow this, since it didn't seem worth making this an error.

However we do error out if you try to add too many plots [the error message isn't great, but we have overloaded the word "plot" here, particularly if this were a `contour(...)` call] - e.g.

```
>>> plot("data", "data", 2)
>>> plot("model", 1, "model", 2, "model", 3, overplot=True)
...
ArgumentErr: Too many elements for overplot
```

The following, where you try to change the plot layout, is ignored (so the cols and rows arguments are ignored when overplot/contour are given):

```
>>> plot("data", "data", 2, cols=1)
>>> plot("model", "model", 2, rows=5, cols=4,  overplot=True)
```

We could make this an error, but is it worth it?

### CODE REVIEW

**Review by hamogu — APPROVED** [review]

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:345]:
```suggestion
       The number of plots.
```

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:13649]:
```suggestion
            The number of rows or columns (to override split plot).
```

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:13762]:
```suggestion
        # Store the original values in case they are overwritten.
```

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:13773]:
```suggestion
        # to overplot too many plots, but we allow less plots to be
```

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:14040]:
```suggestion
           and cols arguments and the automatic calculation no longer
```

File: sherpa/ui/utils.py
Comment (wmclaugh) [inline sherpa/ui/utils.py:16028]:
```suggestion
           and cols arguments and the automatic calculation no longer
```

File: sherpa/ui/utils.py
Comment (DougBurke) [reply sherpa/ui/utils.py:345]: These are from #2126 so once that has been merged (as I've just fixed them) there we can rebase or fix-up this PR to pick them up.

File: sherpa/ui/utils.py
Comment (DougBurke) [reply sherpa/ui/utils.py:13649]: ditto.

File: sherpa/ui/utils.py
Comment (DougBurke) [reply sherpa/ui/utils.py:13773]: I'm going to wait for #2126 to get merged before fixing the other typos.

**Comment by DougBurke** [comment]
Rebased to address the conflict from the doc fixes made in #2126 - some of which were noted here - and to address the remaining doc issues thaw @wmclaugh had noted (and I took the time to scrub both the sherpa/ui/utils.py and sherpa/astro/ui/utils.py to replace "no-longer" with "no longer" and "over-ride" with "override").

**Pull request merged**
